src/OctonetData.cpp

```diff
@@ -256,7 +258,19 @@ bool OctonetData::LoadEPG(void)
 
 void OctonetData::Process()
 {
-  return;
+  // If we have not loaded EPG prior to a 10 second delay after startup
+  // Then request then ask Kodi to request it for each channel
+  std::this_thread::sleep_for(std::chrono::milliseconds(10000));
+
+  if (!m_readInitialEPG)
+  {
+    kodi::Log(ADDON_LOG_INFO, "No request detected from Kodi to load EPG, so requesting now");
```

Comment:
Q: Have we ever encountered this during our tests?

Reply:
you mean EPG not being loaded on startup? Yes. I had to clear PVR data in order to see the HTTP request for the EPG in the logs. So despite not returning EPG for the current selection of unicast channels (which seems to be working now on the octonet) the add-on (or KODI) didn't even try to fetch it most of the time

Reply:
> the add-on (or KODI) didn't even try to fetch it most of the time

IIRC kodi remembers the time stamp the EPG was obtained last time successfully and fetches EPG next time the EPG fetch interval is over (it's a setting).

Could it be (need to check) that even if the octonet does not return any EPG data (for whatever reason), pvr.octonet returns success to Kodi?

Reply:
https://github.com/DigitalDevices/pvr.octonet/blob/Omega/src/OctonetData.cpp#L354

Yes, GetEPGForChannel always returns "success".

Reply:
in my case it returned data, but the data could not be mapped to any channel. So while it had data, it also didn't in a way. No idea which state got reported back to KODI

Reply:
> No idea which state got reported back to KODI

Always "success". GetEPGForChannel never reports any error. For me the question remains, what the actual root cause for the bug is. The load after ten secs is just a golden hammer, not a rel fix. We did not even understand what actually happened.

Reply:
I've removed this commit now.